token-swap/program/src/curve.rs

```diff
@@ -134,18 +144,47 @@ pub trait CurveCalculator: Debug + DynPack {
         swap_destination_amount: u64,
     ) -> Option<SwapResult>;
 
-    /// Get the supply of a new pool (can be a default amount or calculated
-    /// based on parameters)
-    fn new_pool_supply(&self) -> u64;
+    /// Calculate the withdraw fee in pool tokens
+    /// Default implementation assumes no fee
+    fn owner_withdraw_fee(&self, _pool_tokens: u64) -> Option<u64> {
+        Some(0)
+    }
+
+    /// Calculate the pool token equivalent of the owner fee on trade
+    fn owner_fee_to_pool_tokens(
+        &self,
+        owner_fee: u64,
+        trading_token_amount: u64,
+        pool_supply: u64,
+        tokens_in_pool: u64,
+    ) -> Option<u64> {
+        pool_supply
+            .checked_mul(owner_fee)?
+            .checked_div(trading_token_amount)?
+            .checked_div(tokens_in_pool)
+    }
+
+    /// Get the supply for a new pool
+    /// The default implementation is a Balancer-style fixed initial supply
+    fn new_pool_supply(&self) -> u64 {
+        INITIAL_SWAP_POOL_AMOUNT
+    }
 
-    /// Get the amount of liquidity tokens for pool tokens given the total amount
-    /// of liquidity tokens in the pool
-    fn liquidity_tokens(
+    /// Get the amount of trading tokens for the given amount of pool tokens,
+    /// provided the total trading tokens and supply of pool tokens.
+    /// The default implementation is a simple ratio calculation for how many
+    /// trading tokens correspond to a certain number of pool tokens
+    fn pool_tokens_to_trading_tokens(
```

Comment:
Justin and I were discussing the terminology to use here, since "liquidity token" is ambiguous.  I've switched to "trading token" in this PR